torch_object_detection_segmenter.py

```diff
@@ -1,31 +1,202 @@
 __copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
-from typing import Optional
+from typing import Dict, List, Union, Tuple
+import logging
 import numpy as np
-
-from jina import Executor, DocumentArray, requests
+from jina import Document, DocumentArray, Executor
 
 
 class TorchObjectDetectionSegmenter(Executor):
     """
-    Encode the Document blob into embedding.
-
-    :param embedding_dim: the output dimensionality of the embedding
+    :class:`TorchObjectDetectionSegmenter` detects objects
+    from an image using `torchvision detection models`
+    and crops the images according tothe detected bounding boxes
+    of the objects with a confidence higher than a threshold.
+    :param model_name: the name of the model. Supported models include
+        ``fasterrcnn_resnet50_fpn``, ``maskrcnn_resnet50_fpn`
+    :param channel_axis: the axis id of the color channel,
+        ``-1`` indicates the color channel info at the last axis
+    :param confidence_threshold: confidence value from which it
+        considers a positive detection and therefore the object detected will be cropped and returned
+    :param label_name_map: A Dict mapping from label index to label name, by default will be
+        COCO_INSTANCE_CATEGORY_NAMES
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+        TODO: Allow changing the backbone
     """
+    COCO_INSTANCE_CATEGORY_NAMES = [
+        '__background__', 'person', 'bicycle', 'car', 'motorcycle', 'airplane', 'bus',
+        'train', 'truck', 'boat', 'traffic light', 'fire hydrant', 'N/A', 'stop sign',
+        'parking meter', 'bench', 'bird', 'cat', 'dog', 'horse', 'sheep', 'cow',
+        'elephant', 'bear', 'zebra', 'giraffe', 'N/A', 'backpack', 'umbrella', 'N/A', 'N/A',
+        'handbag', 'tie', 'suitcase', 'frisbee', 'skis', 'snowboard', 'sports ball',
+        'kite', 'baseball bat', 'baseball glove', 'skateboard', 'surfboard', 'tennis racket',
+        'bottle', 'N/A', 'wine glass', 'cup', 'fork', 'knife', 'spoon', 'bowl',
+        'banana', 'apple', 'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza',
+        'donut', 'cake', 'chair', 'couch', 'potted plant', 'bed', 'N/A', 'dining table',
+        'N/A', 'N/A', 'toilet', 'N/A', 'tv', 'laptop', 'mouse', 'remote', 'keyboard', 'cell phone',
+        'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'N/A', 'book',
+        'clock', 'vase', 'scissors', 'teddy bear', 'hair drier', 'toothbrush'
+    ]
 
-    def __init__(self, embedding_dim: int = 128, *args, **kwargs ):
+    def __init__(self, model_name: str = None,
+                 on_gpu: bool = False,
+                 channel_axis: int = 0,
+                 confidence_threshold: float = 0.0,
+                 label_name_map: Dict[int, str] = None,
```

Comment:
if default is None, `Optional` type hint

Reply:
Alright